bget: support multiple allocators

lorc opened the pull request:
Hi! 

This PR is needed mostly for virtualization support. It moves all bget state into struct, so we can use multiple bget allocators simultaneously.

jenswi-linaro commented:
I don't think we need to move structure definitions to bget.h, all that is already available to bget_malloc.c due to unorthodox including. I do agree that we need something like bpoolset instead of global variables.

lorc commented:
Hi @jenswi-linaro, 

I agree that right now this isn't needed. But my next action is to introduce new poolset  (`kmalloc`) which is needed for virtualization. So, I'll be forced structure definitions to bget.h anyways.

I can drop that patch from this PR and move it to the next one, if it is more suitable.

jenswi-linaro commented:
Wouldn't it be possible to implement `kmalloc()` or whatever in `bget_malloc.c`?
I don't expect any updates to the upstream bget, but I'd anyway like to avoid unnecessary changes.

lorc commented:
I planed to make `bget_malloc.c` more generic (maybe even rename it to `bget_alloc.c`) and implement `malloc` and `kmalloc`  in different source files.

But, actually, your idea of putting all that in `bget_malloc.c` is quite good. I think, I'll do in this way. This surely will reduce amount of needed changes.

lorc commented:
Pushed new version of this PR. Now there is only one patch, that moves global variables to a structure.
Also, fixed issues with debug build.

Hikey failed at test 1006. But  I can't reproduce it on my setup.

jenswi-linaro commented on lib/libutils/isoc/bget_malloc.c:
```diff
@@ -173,23 +173,25 @@ struct malloc_pool {
 	size_t len;
 };
 
+static struct bpoolset malloc_poolset;
```
This could be statically initialized just as `freelist` was before.

lorc commented:
Adresed comment.

jenswi-linaro commented on lib/libutils/isoc/bget.c:
```diff
@@ -568,10 +568,12 @@ static bufsize pool_len = 0;	      /* 0: no bpool calls have been made
 
 #define ESent	((bufsize) (-(((1L << (sizeof(bufsize) * 8 - 2)) - 1) * 2) - 2))
 
+
```
Line not needed

lorc commented:
Squashed, rebased, added tag.

Removed extra line in resulting commit. It was introduced by adding and then removing code in review commit. So, every commit individually looked okay, but cumulatively they added that extra line.

jforissier commented:
@lorc 
> Hikey failed at test 1006. But I can't reproduce it on my setup.

FYI, was caused by your branch lacking recent commit b0961f98 but xtest already checking it in the CI loop. Now it's rebased and all is good as expected.

Thanks!
